src/test/scala/org/broadinstitute/dsde/workbench/leonardo/dao/MockGoogleDataprocDAO.scala

```diff
@@ -1,20 +1,31 @@
 package org.broadinstitute.dsde.workbench.leonardo.dao
 
+import java.io.File
 import java.util.UUID
 
-import org.broadinstitute.dsde.workbench.leonardo.model.{Cluster, ClusterRequest, ClusterResponse}
+import org.broadinstitute.dsde.workbench.leonardo.config.DataprocConfig
+import org.broadinstitute.dsde.workbench.leonardo.model.ModelTypes.GoogleProject
+import org.broadinstitute.dsde.workbench.leonardo.model._
 
 import scala.collection.concurrent.TrieMap
 import scala.collection.mutable
 import scala.concurrent.{ExecutionContext, Future}
 
-class MockGoogleDataprocDAO extends DataprocDAO {
+class MockGoogleDataprocDAO(protected val dataprocConfig: DataprocConfig) extends DataprocDAO {
+
+  private val clusters: mutable.Map[String, Cluster] = new TrieMap()  // Cluster Name and Cluster
+  private val firewallRules: mutable.Map[GoogleProject, String] = new TrieMap()  // Google Project and Rule Name
+  private val buckets: mutable.Map[GoogleProject, String] = new TrieMap()  // Google Project and Bucket Name ???
+  private val bucketObjects: mutable.Map[String, String] = new TrieMap()   // Bucket Name and File Name
 
```

Comment:
I'm not sure if my approach to setting up the mock dao is correct.

We want to be able to test whether the service adds the firewall, the bucket, and the bucket objects. Should I make these not be private so that we can actually get them in the test to see if the firewall/bucket/objects have been created?

Reply:
yes! akin to how we test our service layer works correctly by checking to see what it did in the database. here we're testing our service layer by checking to see what it asked from (mock) google

Reply:
Yeah, non private fields is fine for the mock DAO. Also `TrieMap` is a concurrent map (I had to look it up), so that's good.